shared/src/main/scala/async/futures.scala

```diff
@@ -363,7 +362,7 @@ enum TaskSchedule:
 class Task[+T](val body: (Async, AsyncOperations) ?=> T):
```

Comment:
It feels reasonable for consistency that the body takes `Async.Spawnable` too (but no `AsyncOperations`). In this case, I think we should wrap all `body` executions in `schedule` with `Async.group` though to ensure structured concurrency compliance.

Reply:
> In this case, I think we should wrap all body executions in schedule with Async.group though to ensure structured concurrency compliance.

when we `run` the Task it's inside a Future, so scoping works. If I were to make it blocking though I would `.group` it.

Reply:
I agree that Tasks, as encapsulated execution plans, should not work on the `Spawn` of the `run`-caller. I would leave it to the Task author, though, to `.group` when it's needed.